pkg/usagestats/reporter.go

```diff
@@ -103,7 +100,7 @@ func agentSeedFileName() string {
 }
 
 // Start inits the reporter seed and start sending report for every interval
-func (rep *Reporter) Start(ctx context.Context) error {
+func (rep *Reporter) Start(ctx context.Context, metrics map[string]interface{}) error {
```

Comment:
`metrics` is going to be static throughout the lifetime of the agent, which wouldn't necessarily represent the actual state of the agent at the time a report is generated. Can we change `metrics` into `func() map[string]interface{}` and call it every time we generate a report instead?